### PR TITLE
update rubocop to 0.42.0

### PR DESCRIPTION
**Pull request by tmaher**
This PR updates Rubocop and fixes a small handful of cops & changes.

* `Style/TrailingComma` got broken up into `Style/TrailingCommaInLiteral` and `Style/TrailingCommaInArguments` (see https://github.com/bbatsov/rubocop/issues/2678)
* `Style/SingleSpaceBeforeFirstArg` was renamed `Style/SpaceBeforeFirstArg` (see https://github.com/bbatsov/rubocop/issues/2504#issuecomment-175162564)
* `Style/SymbolProc` and `Style/GuardClause` are fixed repo-wide, and the exceptions removed from `.rubocop_todo.yaml`

/cc @jch @mtodd

### CODE REVIEW

**Comment by jch** [comment]
Hmmm not sure where those CI failures are coming from. @davesims do they look familiar?

**Comment by davesims** [comment]
@jch Nope, haven't seen that. ¯\_(ツ)_/¯

File: lib/net/ldap.rb [hunk 1]
```diff
@@ -1286,11 +1286,9 @@ def use_connection(args)
     else
       begin
         conn = new_connection
-        if (result = conn.bind(args[:auth] || @auth)).result_code == Net::LDAP::ResultCodeSuccess
-          yield conn
-        else
-          return result
-        end
+        result = conn.bind(args[:auth] || @auth)
+        return result unless result.code == Net::LDAP::ResultCodeSuccess
```
Comment (satoryu) [inline lib/net/ldap.rb]: rubocode failed to convert codes. `result_code` should be called here, not `code`. All tests would be passed by fixing it.  @davesims @jch @tmaher 😃

Comment (satoryu) [inline lib/net/ldap.rb]: @tmaher I have just opened new PR to this your branch to fix this bug tmaher/ruby-net-ldap#1 . please check it.

**Comment by tmaher** [comment]
@satoryu - Yep, that fixed it. Sorry about that.

File: lib/net/ldap.rb (hunk 1 above)
Comment (jch) [inline lib/net/ldap.rb]: Thanks @satoryu, nice catch

**Pull request merged**